velox/functions/lib/Re2Functions.cpp

```diff
@@ -738,8 +744,9 @@ std::shared_ptr<exec::VectorFunction> makeLike(
       name,
       inputArgs[1].type->toString());
 
+  bool hasEscape = numArgs == 3;
   char escapeChar;
```

Comment:
If you used a `std::optional` here, you wouldn't need any other changes to this function.